src/opentimelineio/deserialization.cpp

```diff
@@ -37,10 +37,13 @@ class JSONDecoder : public OTIO_rapidjson::BaseReaderHandler<OTIO_rapidjson::UTF
     
     bool Null() {  return store(any()); }
     bool Bool(bool b) { return store(any(b)); }
-    bool Int(int i) {  return store(any(i)); }
-    bool Uint(unsigned u) {  return store(any(int(u))); }
-    bool Int64(int64_t i) { return store(any(i)); }
-    bool Uint64(uint64_t u) { return store(any(int64_t(u))); }
+
+    // coerce all integer types to int64_t
+    bool Int(int i) {  return store(any(static_cast<int64_t>(i))); }
+    bool Uint(unsigned u) {  return store(any(static_cast<int64_t>(u))); }
+    bool Int64(int64_t i) { return store(any(static_cast<int64_t>(i))); }
+    bool Uint64(uint64_t u) { return store(any(static_cast<int64_t>(u))); }
```

Comment:
if we change this to 

```cpp
bool Uint64(uint64_t u) { return store(any(static_cast<int64_t>(u & 0x7FFFFFFFFFFFFFFF))); }
```
we will avoid undefined behavior for values greater than 2^63, by defining the behavior to be that we don't allow UB overflow states.. This doesn't matter now in a practical sense, but it will show up when we start fuzz testing.